docs_src/design/adr/core/0019-EdgeX-CLI-V2.md

```diff
@@ -0,0 +1,106 @@
+
+Other issues with the existing client include:
+
+- lack of tab completion
+- default output of commands is too verbose
+- verbose output sometime prevents use of jq 
+- static configuration file required (i.e. no registry support)
+- project hierarchy not conforming to best practice guidelines 
+
+## History
+The original Hanoi V1 client was created by a team at VMWare which is no longer participating in the project. 
+Canonical will lead the development of the Ireland/Jakarta V2 client.
+
+## Decision
+
+1. Use standardized command-line args/flags
+
+| Argument/Flag      | Description |
+| ----------- | ----------- |
+| `-d`, `--debug`      | show additional output for debugging purposes (e.g. REST URL, request JSON, …). This command-line arg will replace -v, --verbose and will no longer trigger output of the response JSON (see -j, --json).       |
+| `-j`, `--json`   | output the raw JSON response returned by the EdgeX REST API and *nothing* else. This output mode is used for script-based usage of the client.    |
+| `--version`   | output the version of the client and if available, the version of EdgeX installed on the system (using the version of the metadata data service)   |
+
+
+2. Restructure the Go code hierarchy to follow the [most recent recommended guidelines](https://github.com/golang-standards/project-layout). For instance /cmd should just contain the main application for the project, not an implementation for each command - that should be in /internal/cmd
+
+3. Take full advantage of the features of the underlying command-line library, [Cobra](https://github.com/spf13/cobra), such as tab-completion of commands.
+
+4. Add support for the four common endpoints - `/ping`, `/metrics`, `/version`, `/status`
+
+5. Allow overlap of command names across services by supporting an argument to specify the service to use: `-m/--metadata`, `-c/--command`, `-n/--notification`, `-s/--scheduler` or `--data` (which is the default). Examples:
+
+    - `edgex-cli ping --data`
+    - `edgex-cli ping -m`
+    - `edgex-cli version -c`
+
+6. Implement all required V2 endpoints for core services
+
+    **Core Command** (`edgex-cli (c)ommand`)
+    - **`command`** `read | write | list`
+
+    **Core Data** (`edgex-cli (d)ata`)
+    - **`event`** `add | count | list | rm | scrub**`
+    - **`reading`** `count | list`
+
+    **Metadata** (`edgex-cli (m)etadata`)
+    
+    - **`device`**  `add | adminstate | list | operstate | rm | update`
+    - **`deviceprofile`**  `add | list | rm | update`
+    - **`deviceservice`** ` add | list | rm | update`
+    - **`provisionwatcher`**  `add | list | rm | update`
+    
+    **Support Notifications** ( `edgex-cli (n)otifications`)
+    - **`notification`** `add | list | rm`
+    - **`subscription`** `add | list | rm`
+
+   **Support Scheduler** ( `edgex-cli (s)cheduler`)
+    - **`interval`** `add | list | rm | update` 
+
+    **Common endpoints in all services**
+    - **`version`**
```

Comment:
Aren't these already mentioned in `#4` above? Maybe just delete it, and leave these here in this section?

Reply:
deleted.